Tests/src/PublicTestBase.cpp

```diff
@@ -36,12 +37,20 @@ void PublicTestBase::SetUp()
 		});
 
 	csp::systems::SystemsManager::Get().GetLogSystem()->LogMsg(csp::systems::LogLevel::Verbose, "Foundation initialised!");
+
+	auto Connection = csp::systems::SystemsManager::Get().GetMultiplayerConnection();
+
+	AWAIT(Connection, SetAllowSelfMessagingFlag, false);
 }
 
 void PublicTestBase::TearDown()
 {
 	::testing::Test::TearDown();
 
+	auto Connection = csp::systems::SystemsManager::Get().GetMultiplayerConnection();
+
+	AWAIT(Connection, SetAllowSelfMessagingFlag, false);
```

Comment:
Why do we set it false again here in teardown, won't it already be false?

Reply:
Isn't the multiplayer connection object destroyed (along with CSP itself) when CSP is [deinitialised ](https://github.com/magnopus-opensource/connected-spaces-platform/blob/develop/Tests/src/PublicTestBase.cpp#L55) at the end of each test?

Reply:
Right, I think I get it now.

Next stupid question - why do we need to set it to false in teardown if we also set it to true before the start of each test? Won't that also cover the case where a test sets it to true and forgets to set it back to false?

Reply:
@MAG-AW pointed out to me on a call that:
* This flag persists across user sessions
* And so setting it to false at the end of the test also ensures it gets set back to the expected value after the last test runs.